scripts/globals/job_utils/rune_fencer.lua

```diff
@@ -50,8 +50,163 @@ local function enforceRuneCounts(target)
+        return 0
+    end
+
+    local runeStatMap = {
+                            [xi.effect.IGNIS]  = xi.mod.STR,
+                            [xi.effect.GELUS]  = xi.mod.DEX,
+                            [xi.effect.FLABRA] = xi.mod.VIT,
+                            [xi.effect.TELLUS] = xi.mod.AGI,
+                            [xi.effect.SULPOR] = xi.mod.INT,
+                            [xi.effect.UNDA]   = xi.mod.MND,
+                            [xi.effect.LUX]    = xi.mod.CHR,
+                        }
+
+    local stat = runeStatMap[effect:getType()]
+    if stat ~= nil then
+        return math.floor(target:getStat(stat) * 0.5)
+    end
+
+    return 0
+end
+
+-- source https://www.bg-wiki.com/ffxi/Vivacious_Pulse
+local function calculateVivaciousPulseHealing(target)
+
+    local divineMagicSkillLevel = target:getSkillLevel(xi.skill.DIVINE_MAGIC)
+    local HPHealAmount = 10 + math.floor(divineMagicSkillLevel / 2 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) -- Bonus of 1-20%  from Vivacious pulse job points.
+    local tenebraeRuneCount = 0
+    local bonusPct = 1.0 -- todo, add in bonus from augment attached to Futhark Claymore, Peord Claymore, and Morgelai. Need extdata for proper values.
+
+    local debuffs = {}
+    local debuffCount = 0
+
+    local effects = target:getStatusEffects()
+    for _, effect in ipairs(effects) do
+        local type = effect:getType()
+        if type >= xi.effect.IGNIS and type <= xi.effect.TENEBRAE then
+            HPHealAmount = HPHealAmount + getRuneHealAmount(effect, target)
+        elseif
+            type == xi.effect.POISON or type == xi.effect.PARALYSIS or
+            type == xi.effect.BLINDNESS or type == xi.effect.SILENCE or type == xi.effect.MUTE or type == xi.effect.CURSE_I or
+            type == xi.effect.CURSE_II or type == xi.effect.DOOM or type == xi.effect.VIRUS or
+            type == xi.effect.PLAGUE or type == xi.effect.PETRIFICATION
+        then
+            debuffs[debuffCount+1] = type
+        debuffCount = debuffCount + 1
+        end
+
+        if type == xi.effect.TENEBRAE then -- runes that also restore MP
+            tenebraeRuneCount = tenebraeRuneCount + 1
+        end
+    end
+
+    if tenebraeRuneCount > 0 then -- only restore MP if there's one or more tenebrae rune active
+        local MPHealAmount = math.floor(divineMagicSkillLevel / 10 * (100 + target:getJobPointLevel(xi.jp.VIVACIOUS_PULSE_EFFECT)) / 100) * (tenebraeRuneCount + 1)
+        target:addMP(MPHealAmount) -- augment bonusPct does not apply here according to testing.
+    end
+
+    if debuffCount > 0 and (target:getEquipID(xi.slot.HEAD) == 26782 or target:getEquipID(xi.slot.HEAD) == 26783) then -- add random removal of Poison, Paralyze, Blind, Silence, Mute, Curse, Bane, Doom, Virus, Plague, Petrification via AF3 head (source: https://www.bg-wiki.com/ffxi/Erilaz_Galea)
```

Comment:
Magic numbers for Item ID should probably be added to the enum in scripts/globals/items.lua

Reply:
mods!